### PR TITLE
Better SVG output

### PR DESCRIPTION
**Pull request by BagelOrb**
All the debug fun with zero risk!

I use this class a lot for debugging and I don't like going back in debugging functionality after a project, so please incorporate this improved debugging output functionality.

There is literally 0% risk.

Whats new? : 
- RGB color control
- some more aux functions
- accurate scaling, rather than exact canvas size
- bug fixes relating to output canvas size
- removed canvas border related to above to features
- output SVG layers (compatible with Inkscape)

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Developers, see issue CURA-6969. Giggity.

File: src/utils/SVG.cpp [hunk 1]
```diff
@@ -210,7 +285,45 @@ void SVG::writePolygon(ConstPolygonRef poly, Color color, float stroke_width)
     int i = 0;
     for (Point p1 : poly)
     {
-        if (color == Color::RAINBOW)
+        if (color.color == Color::RAINBOW)
+        {
+            int g = (i * 255 * 11 / size) % (255 * 2);
+            if (g > 255) g = 255 * 2 - g;
+            int b = (i * 255 * 5 / size) % (255 * 2);
+            if (b > 255) b = 255 * 2 - b;
```
Comment (Ghostkeeper) [inline src/utils/SVG.cpp]: Brackets on new line.
```suggestion
            if (b > 255)
            {
                b = 255 * 2 - b;
            }
```

File: src/utils/SVG.cpp [hunk 2]
```diff
@@ -210,7 +285,45 @@ void SVG::writePolygon(ConstPolygonRef poly, Color color, float stroke_width)
     int i = 0;
     for (Point p1 : poly)
     {
-        if (color == Color::RAINBOW)
+        if (color.color == Color::RAINBOW)
+        {
+            int g = (i * 255 * 11 / size) % (255 * 2);
+            if (g > 255) g = 255 * 2 - g;
```
Comment (Ghostkeeper) [inline src/utils/SVG.cpp]: Brackets on new line.

```suggestion
            if (g > 255)
            {
                g = 255 * 2 - g;
            }
```

File: src/utils/SVG.cpp [hunk 3]
```diff
@@ -210,7 +285,45 @@ void SVG::writePolygon(ConstPolygonRef poly, Color color, float stroke_width)
     int i = 0;
     for (Point p1 : poly)
     {
-        if (color == Color::RAINBOW)
+        if (color.color == Color::RAINBOW)
```
Comment (Ghostkeeper) [inline src/utils/SVG.cpp]: Perhaps it's good to document what this does.

This seems to select one of 11 rainbow colours depending on the number of vertices in the polygon.

File: src/utils/SVG.cpp [hunk 4]
```diff
@@ -51,9 +74,37 @@ SVG::SVG(const char* filename, AABB aabb, Point canvas_size, Color background)
...
+    fprintf(out,"        <dc:type\n");
+    fprintf(out,"           rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\" />\n");
+    fprintf(out,"        <dc:title></dc:title>\n");
+    fprintf(out,"      </cc:Work>\n");
+    fprintf(out,"    </rdf:RDF>\n");
+    fprintf(out,"  </metadata>\n");
+    fprintf(out,"  <g\n");
+    fprintf(out,"    inkscape:groupmode=\"layer\"\n");
+    fprintf(out,"    inkscape:label=\"layer%zu\"\n", layer_nr );
+    fprintf(out,"    id=\"layer%zu\">\n", layer_nr );
```
[the remark this answers is not in the document]
Comment (BagelOrb) [reply src/utils/SVG.cpp]: Verified this to be true. Fixed.

File: src/utils/SVG.cpp (hunk 3 above)
Comment (BagelOrb) [reply src/utils/SVG.cpp]: ok

File: src/utils/SVG.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (BagelOrb) [reply src/utils/SVG.cpp]: I mean: by virtue of the appeal to expert opinion it was true by default - coming from you. Still it's never a bad idea to double check ;)

**Pull request merged**